src/components/DevicesModaleConfigureView.spec.jsx

```diff
@@ -0,0 +1,771 @@
+  return {
+    ...original,
+    updateDirectoriesExclusions: jest.fn(),
+    useFolders: jest.fn()
+  }
+})
+
+describe('DevicesModaleConfigureView', () => {
+  const mockClient = createMockClient({})
+  const mockDevice = {
+    _id: 'mock-device-id',
+    _type: 'io.cozy.oauth.clients',
+    client_kind: 'desktop',
+    client_name: 'Mock Device'
+  }
+  const includedFolders = [
+    { id: 'Administrative', name: 'Administrative', not_synchronized_on: [] },
+    { id: 'Konnectors', name: 'Konnectors', not_synchronized_on: null },
+    { id: 'Accounts', name: 'Accounts', not_synchronized_on: undefined },
+    { id: 'Ciphers', name: 'Ciphers' }
+  ]
+  const excludedFolders = [
+    { id: 'Photos', name: 'Photos', not_synchronized_on: [mockDevice] },
+    { id: 'Videos', name: 'Videos', not_synchronized_on: [mockDevice] }
+  ]
+  const mockFolders = includedFolders.concat(excludedFolders).sort(f => f.name)
+  const mockCancelAction = jest.fn()
+  const mockOnDeviceConfigured = jest.fn()
+
+  const setup = () => {
+    const root = render(
+      <AppLike client={mockClient}>
+        <DevicesModaleConfigureView
+          device={mockDevice}
+          cancelAction={mockCancelAction}
+          onDeviceConfigured={mockOnDeviceConfigured}
+        />
+      </AppLike>
+    )
+    return { root }
+  }
+
+  afterEach(() => {
+    updateDirectoriesExclusions.mockClear()
+    mockOnDeviceConfigured.mockClear()
+    mockCancelAction.mockClear()
+  })
+
+  describe('when the folders query is still loading', () => {
+    beforeEach(() => {
+      useFolders.mockReturnValue({
+        loading: true,
+        failed: false,
+        folders: []
+      })
+    })
+
+    it('should render an loading spinner', () => {
+      const { root } = setup()
+      expect(
+        root.queryByText(
+          'Select the folders to synchronize with your computer:'
+        )
+      ).not.toBeInTheDocument()
+      expect(root.getByRole('progressbar')).toBeVisible()
+    })
+
+    describe('and the dialog is validated', () => {
+      const run = () => {
+        const { root } = setup()
+        fireEvent.click(root.getByRole('button', { name: 'OK' }))
+      }
+
+      it('should call updateDirectoriesExclusions without folders', async () => {
+        run()
+
+        expect(updateDirectoriesExclusions).toHaveBeenCalledTimes(1)
+        expect(updateDirectoriesExclusions).toHaveBeenCalledWith({
+          client: mockClient,
+          deviceToConfigure: toCozyOAuthClient(mockDevice),
+          foldersToInclude: [],
+          foldersToExclude: []
+        })
+        await waitFor(() =>
+          expect(mockOnDeviceConfigured).toHaveBeenCalledTimes(1)
+        )
+      })
+    })
+  })
+
+  describe('when the folders query failed', () => {
+    beforeEach(() => {
+      useFolders.mockReturnValue({
+        loading: false,
+        failed: true,
+        folders: mockFolders // XXX: just checking that this has no effects on the UI
```

Comment:
What's the meaning of XXX?

Reply:
It's a common pattern to catch the eye (my editor highlights it for example). You can see others are using it https://stackoverflow.com/questions/1452934/what-is-the-meaning-of-xxx-in-code-comments but we use it at least in the stack and desktop code bases.
I use it as a note when explaining something that could seem rather odd.